File: src/main/java/sk/hudak/prco/task/SubmitTask.java
package sk.hudak.prco.task;

import lombok.Getter;
import sk.hudak.prco.api.EshopUuid;

@Getter
public abstract class SubmitTask<T, K> {

    public abstract void doInTask(EshopUuid eshopUuid, T param1, K param2) throws Exception;
}

File: src/main/java/sk/hudak/prco/task/SubmitTask.kt
package sk.hudak.prco.task;

import sk.hudak.prco.api.EshopUuid;

@FunctionalInterface
public interface SubmitTask<T, K> {

    void doInTask(EshopUuid eshopUuid, T param1, K param2) throws Exception;
}
